sdk/containerregistry/container-registry/test/public/utils.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ClientSecretCredential } from "@azure/identity";
+import { env, RecorderEnvironmentSetup } from "@azure/test-utils-recorder";
+import { ContainerRegistryClient, ContainerRepositoryClient } from "../../src";
+
+// When the recorder observes the values of these environment variables in any
+// recorded HTTP request or response, it will replace them with the values they
+// are mapped to below.
+const replaceableVariables: Record<string, string> = {
+  CONTAINER_REGISTRY_ENDPOINT: "https://myregistry.azurecr.io",
+  AZURE_TENANT_ID: "azure_tenant_id",
+  AZURE_CLIENT_ID: "azure_client_id",
+  AZURE_CLIENT_SECRET: "azure_client_secret",
+  SUBSCRIPTION_ID: "subscription_id"
+};
+
+export const recorderEnvSetup: RecorderEnvironmentSetup = {
+  // == Recorder Environment Setup == Add the replaceable variables from
+  // above
+  replaceableVariables,
+
+  // We don't use this in the template, but if we had any query parameters
+  // we wished to discard, we could add them here
+  queryParametersToSkip: [],
+
+  // Finally, we need to remove the AAD `refresh_token` from any requests.
+  // This is very important, as it cannot be removed using environment
+  // variable or query parameter replacement.  The
+  // `customizationsOnRecordings` field allows us to make arbitrary
+  // replacements within recordings.
+  customizationsOnRecordings: [
+    (recording: string): string =>
+      recording.replace(/"refresh_token":"[^"]*"/g, `"refresh_token":"refresh_token"`),
```

Comment:
Looking good!

Should we..
- scope it down as was done in the access_token PR for the recorder? (Currently, this would match everywhere in the recording, which is still fine I guess)
- move it to the recorder?

Reply:
> * scope it down as was done in the access_token PR for the recorder?

I remember we had discussion to limit it to `.reply()` but I don't remember what requires us to do so.  Could you please remind me?

Reply:
This I would say is not important.

Nothing requires us to do so(as of now), I just meant it as future-proofing(far-fetched).
(I did scope access_tokens in my PR for `reply()` and made sure the content-type to be application-json.)

Reply:
Probably will do it after release this. Logged https://github.com/azure/azure-sdk-for-js/issues/14695